lib/tls.js

```diff
@@ -143,9 +146,13 @@ exports.convertALPNProtocols = function convertALPNProtocols(protocols, out) {
   // If protocols is Array - translate it into buffer
   if (ArrayIsArray(protocols)) {
     out.ALPNProtocols = convertProtocols(protocols);
-  } else if (isArrayBufferView(protocols)) {
+  } else if (Buffer.isBuffer(protocols) || isUint8Array(protocols)) {
     // Copy new buffer not to be modified by user.
     out.ALPNProtocols = Buffer.from(protocols);
+  } else if (isArrayBufferView(protocols)) {
+    out.ALPNProtocols = Buffer.from(protocols.buffer.slice(),
```

Comment:
Is copying necessary?

Reply:
From the comment above the answer seems to be, yes.

Reply:
Yup. 😄